tests/test_node.py

```diff
@@ -371,3 +383,12 @@ def test_apply_before_node_is_ready():
 
     with pytest.warns(RuntimeWarning):
         Node(style=style, applicator=applicator)
+
+
+def test_applicator_has_node_reference():
```

Comment:
It took me a moment to work out why this test was different to the last assertion in `test_apply_before_node_is_ready`, except that it wasn't actually asserting that the warning was raised. A comment on the fact that AttributeTestStyle asserts that it is re-applied would be helpful.